sdk/signalr/Microsoft.Azure.WebJobs.Extensions.SignalRService/src/Config/ServiceHubContextStore.cs

```diff
@@ -55,9 +65,30 @@ private async Task<IServiceHubContext> GetFromLazyAsync(string hubName, (Lazy<Ta
             }
             catch (Exception)
             {
+                // Allow to retry for transient errors.
                 _store.TryRemove(hubName, out _);
                 throw;
             }
         }
+
+        public void Dispose()
+        {
+            _serviceManager.Dispose();
+            foreach (var tuple in _store.Values)
+            {
+                tuple.Value?.DisposeAsync().Wait();
```

Comment:
Why not use IAsyncDisposable? Pretty sure that's available for SDK libs since others use it too

Reply:
I just tried to use the `IAsyncDisposable` from `Microsoft.Bcl.AsyncInterfaces` packages, but found that the `ServiceProvider` won't dispose such a type implemented `IAsyncDisposable` during disposing. I am not sure I am using it correctly.

Reply:
When you register those on DI you need to use the async scope

https://devblogs.microsoft.com/dotnet/announcing-net-6/#microsoft-extensions-dependencyinjection-createasyncscope-apis

Reply:
The service is singleton and it shouldn't need the async scope. The problem is not an `InvalidOperationException` but that the `IAsyncDisposable` alone isn't call when the host is disposed. Currently I have implemented both methods.